### ISSUE TITLE
Ordering modules bug

### ISSUE THREAD

**Issue by niosme** [joomla/joomla-cms#44545]
### Steps to reproduce the issue

When we have our modules shorted, we filter by position e.g. footer and then we enter in some modules and change the name , permissions or something is changed and we save it the order is changing.

https://www.awesomescreenshot.com/video/34097008?key=8514d5e511a9179ccb0ccc08ac81d171

### Expected result

Keep the ordering as it was if we dont change it.

### Actual result

Ordering changes and this have result in the frontend.

### System information (as much as possible)

Joomla 5.2.2

### Additional comments

When we just logged in and we do this even the outer drag to position doesnt work. In the video it worked fine. I had to play a little on the modules list to make the drag to sort work fine.

**Comment by fgsw** [comment]
@niosme Can you write an example of your Issue? I have watched your video 2 times and didn't get the point, where your Issue happen.

**Comment by niosme** [comment]
> [@niosme](https://github.com/niosme) Can you write an example of your Issue? I have watched your video 2 times and didn't get the point, where your Issue happen.

When i enter a module and change something in it like the name, we hit save and close and the ordering changes as we show in the list.
I ll take another video more clear.

**Comment by niosme** [comment]
Here is a more clear video:

https://www.awesomescreenshot.com/video/34097008?key=8514d5e511a9179ccb0ccc08ac81d171

After a long time of playing and just changing the names it works fine. But the most time the order is changing.

**Comment by niosme** [comment]
> [@niosme](https://github.com/niosme) Can you write an example of your Issue? I have watched your video 2 times and didn't get the point, where your Issue happen.

Here is a more clear video:

https://www.awesomescreenshot.com/video/34097008?key=8514d5e511a9179ccb0ccc08ac81d171

After a long time of playing and just changing the names it works fine. But the most time the order is changing.

**Comment by fgsw** [comment]
@niosme Did you get your Issue on a clean Installation (without 3pd-Modules like "Filter - Tags" showing at 00:25 in your second video)?

**Comment by Fedik** [comment]
I going to set it as a bug. 
I have the same problem sometimes, however it happens very randomly, and hard to reproduce.

If someone can discover a clear pattern when and why it happening, that would be very helpful.

**Comment by niosme** [comment]
> [@niosme](https://github.com/niosme) Did you get your Issue on a clean Installation (without 3pd-Modules like "Filter - Tags" showing at 00:25 in your second video)?

Yes even with the default and without third party modules this occurs.

It always happen to me if i filter by position and then try to save it because its easily visible. The bad thing is the frontend which changes.

But also have occured event without filter. But its easier to check it if you filter by position and you have not a lot of items.
If you start playing a lot after a lot of saves it works fine. Like everything take its position and then its fine.

I will check the database changes and the form data sended like ordering that is going to be saved when i find some time to understand the reason and help with this.

Also the ordering on drag n scroll from the list have the same issue sometimes. If we change the order and refresh the page the ordering is not what we setted visually.

Ill try to figure it out.

**Comment by pixelhexe** [comment]
@niosme This problem sounds familiar to me. In your video on the modules overview there is the column "Pages" with values all "Selected". 

Do the modules on the position "top-a" belong all to the same page(s)? Or what happens if you filter the page where your module is included BEFORE you edit your module? Filter options on the modules overview: "- Select Menu Item -"? 

I remember that the order of modules only works properly in combination with the suitable menu item.

**Comment by reDimDev** [comment]
I forgot to add downlaod links to the images.

[https://redim.de/media/downloads/appendix_a.zip](https://redim.de/media/downloads/appendix_a.zip "Appendix A")

[https://redim.de/media/downloads/appendix_b.zip](https://redim.de/media/downloads/appendix_b.zip "Appendix B")<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44545">issues.joomla.org/tracker/joomla-cms/44545</a>.</sub>

**Comment by richard67** [comment]
If you are on Joomla 5 and are using MariaDB 10.x (10.2.0 or newer, but not 11.x):

Could you test the following framework PR and leave a comment there about the result?

https://github.com/joomla-framework/database/pull/335

Thanks in advance.
